File: src/main/java/org/springframework/samples/petclinic/JettyHandler.java
package org.springframework.samples.petclinic;

import org.eclipse.jetty.server.Request;
import org.eclipse.jetty.server.handler.AbstractHandler;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.servlet.ServletOutputStream;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.io.OutputStreamWriter;

import static java.nio.charset.StandardCharsets.UTF_8;

public class JettyHandler extends AbstractHandler {
    private static final Logger logger = LoggerFactory.getLogger(JettyHandler.class);

    @Override
    public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException {
        logger.debug("build 0.3.2; Handle request with target {}", target);
        baseRequest.setHandled(true);
        response.setContentType("text/plain");
        final String content = "Target " + target;
        opaTarget(target);
        response.setContentLength(content.length());
        final ServletOutputStream outputStream = response.getOutputStream();
        OutputStreamWriter writer = new OutputStreamWriter(outputStream, UTF_8);
        writer.write(content);
        writer.flush();
    }

    private void opaTarget(String target) {
        if (target.equals("/opa")) {
            logger.debug("Handle request with opa target {}. sleep...", target);
            waitMsc(200);
        }
    }

    private void waitMsc(int msc) {
        try {
            logger.debug("sleep msc {}...", msc);
            Thread.sleep(msc);
        } catch (InterruptedException e) {
            logger.error(String.valueOf(e));
        }
    }
}

File: src/main/kotlin/org/springframework/samples/petclinic/JettyHandler.kt
package org.springframework.samples.petclinic

import mu.KotlinLogging
import org.eclipse.jetty.server.Request
import org.eclipse.jetty.server.handler.AbstractHandler
import java.io.OutputStreamWriter
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse
import kotlin.text.Charsets.UTF_8

private val logger = KotlinLogging.logger {}

class JettyHandler : AbstractHandler() {
    override fun handle(target: String?, baseRequest: Request, request: HttpServletRequest, response: HttpServletResponse) {
        logger.debug { "Handle a request with the target=$target" }
        baseRequest.isHandled = true
        response.contentType = "text/plain"
        val content = "Target $target"
        response.setContentLength(content.length)
        val outputStream = response.outputStream
        val writer = OutputStreamWriter(outputStream, UTF_8)
        writer.write(content)
        writer.flush()
    }

}
